static/js/unread_ops.js

```diff
@@ -29,6 +29,21 @@ export function mark_all_as_read() {
                 save_compose: true,
             });
         },
+        error(xhr) {
+            if (xhr.status === 504) {
```

Comment:
I'll also need to add a mention of this in the API changelog, once we're set on this approach or another.